integrations/spark/src/main/java/marquez/spark/agent/lifecycle/SparkSQLExecutionContext.java

```diff
@@ -95,6 +103,26 @@ private ParentRunFacet buildParentFacet() {
         .build();
   }
 
+  private DatasetMetricFacet buildMetricFacet(Statistics statistics) {
+    if (statistics != null) {
+      BigInt count =
+              statistics.rowCount()
+                      .getOrElse(new AbstractFunction0<BigInt>() {
+                        @Override
+                        public BigInt apply() {
+                          return BigInt.apply(0);
+                        }
+                      });
+      return DatasetMetricFacet.builder()
+              .metricValues(
+                      ImmutableMap.of("rows", DatasetMetricFacet.Metric.count(count.longValue())))
+              .build();
+
+    } else {
+      return null;
+    }
```

Comment:
I think we should make a facet that is specific to this particular data. 
A generic data quality facet is also useful but used in a different context in complement of this.
How about: 
```
OutputSizeFacet: {
   rowCount: 123,
   ...
}                   
```
Also the logicalPlan probably is an estimate for the planner when the physical plan after execution would probably have the actual count.